llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -9418,8 +9418,14 @@ AArch64TargetLowering::LowerCall(CallLoweringInfo &CLI,
                     *DAG.getContext());
   RetCCInfo.AnalyzeCallResult(Ins, RetCC);
 
-  // Set type id for call site info.
-  if (MF.getTarget().Options.EmitCallGraphSection && CB && CB->isIndirectCall())
+  // Set type id for call site info and metadata 'call_target'.
+  // We are filtering for:
+  // a) The call-graph-section use case that wants to know about indirect
+  //    calls, or
+  // b) We want to annotate indirect calls.
+  if (CB && CB->isIndirectCall() &&
+      (MF.getTarget().Options.EmitCallGraphSection ||
+       MF.getTarget().Options.EmitCallSiteInfo))
```

Comment:
This condition is appearing several times - can it be tested for in fewer/one place and have the consequences flow out more naturally/with fewer special cases spread around? Or, if that's not possible, at least the condition may need to be refactored into a reusable function.